scrnaseq.Rmd

```diff
@@ -1674,142 +1748,400 @@ if (degs_found) {
 
 ### Heatmaps of all differentially expressed genes {.tabset}
 #### Up- and down-regulated genes
-```{r degs_heatmap_all, eval=degs_found, fig.height=20}
-# Let us re-think if we want to limit the number of cells in this plot, since this takes a while. 
-
-if (degs_found) {
+```{r marker_degs_heatmap_all, eval=marker_degs_found, fig.height=20}
+if (marker_degs_found) {
+  # This will sample 500 cells; the number of cells per seurat_cluster will be proportional
+  cells_subset = ScSampleCells(sc, n=500, group="seurat_clusters", group_proportional=TRUE)
+  # With fig.height = 20, 300 features can be shown; distribute among clusters
+  features_per_group = 300/length(levels(marker_degs_filt$all$cluster))
```

Comment:
The problem with the heatmap for me is that it has no rownames. Everyone will ask for this I would assume. Here it might be better to focus on some important genes.

How it is done in publications?

Reply:
No in this case no row names were needed. I'd like to have one overview heatmap, where you see the amount of degs and how it looks like. For huge heatmaps like this, no gene names are needed. 

If we additionally do smaller heatmaps, showing the top 100 for examples, we also need gene names.

Reply:
Done